pinot-core/src/main/java/org/apache/pinot/core/common/DataFetcher.java

```diff
@@ -611,12 +623,8 @@ void readBytesValues(int[] docIds, int length, byte[][] valueBuffer) {
         _dictionary.readBytesValues(dictIdBuffer, length, valueBuffer);
       } else {
         switch (_storedType) {
-          case STRING:
-            for (int i = 0; i < length; i++) {
-              valueBuffer[i] = BytesUtils.toBytes(_reader.getString(docIds[i], readerContext));
-            }
-            break;
           case BYTES:
+          case STRING:
```

Comment:
We can remove the switch and directly read the values

Reply:
I thought we don't want to support this for other types?